### PR TITLE
feat: hide play button when fetch state is completed

### PR DESCRIPTION
**Pull request by YannickPferr**
Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok

What
----

We now only show the play button (and it's functionality) when a statement is not COMPLETED.

Play button visible:
image

Play button not visible:
image

### CODE REVIEW

File: internal/pkg/flink/components/table_view.go [hunk 1]
```diff
@@ -232,19 +232,36 @@ func (t *TableView) JumpDown() {
...
-	playPause := "Play"
+
+	if fetchState == types.Completed {
+		return t.getTableShortcutsForCompletedFetchState(toggleTableModeText)
+	}
+
+	toggleAutoRefreshText := "Play"
 	if isAutoRefreshRunning {
-		playPause = "Pause"
+		toggleAutoRefreshText = "Pause"
```
Comment (Guttz) [inline internal/pkg/flink/components/table_view.go:247]:
```suggestion
		toggleRefreshText = "Pause"
```

File: internal/pkg/flink/components/table_view.go [hunk 2]
```diff
@@ -232,19 +232,36 @@ func (t *TableView) JumpDown() {
...
+	}
+}
+
+func (t *TableView) getTableShortcutsForNonCompletedFetchState(toggleTableModeText, toggleAutoRefreshText string) []types.Shortcut {
 	return []types.Shortcut{
 		{KeyText: ExitTableViewShortcut, Text: "Quit"},
-		{KeyText: ToggleTableModeShortcut, Text: mode},
-		{KeyText: ToggleAutoRefreshShortcut, Text: playPause},
+		{KeyText: ToggleTableModeShortcut, Text: toggleTableModeText},
+		{KeyText: ToggleAutoRefreshShortcut, Text: toggleAutoRefreshText},
```
Comment (Guttz) [inline internal/pkg/flink/components/table_view.go:264]:
```suggestion
		{KeyText: ToggleRefreshShortcut, Text: toggleRefreshText},
```

File: internal/pkg/flink/components/table_view.go [hunk 3]
```diff
@@ -123,7 +123,7 @@ func (t *TableView) RenderTable(tableTitle string, statementResults *types.Mater
 
 	t.infoBar.SetLastRefreshTimestamp(lastRefreshTimestamp)
 	t.infoBar.SetFetchState(fetchState)
-	t.createTableView(NewShortcuts(t.getTableShortcuts(statementResults, isAutoRefreshRunning)))
+	t.createTableView(NewShortcuts(t.getTableShortcuts(statementResults, isAutoRefreshRunning, fetchState)))
```
[the remark this answers is not in the document]
Comment (YannickPferr) [reply internal/pkg/flink/components/table_view.go:126]: Yeah true, now that we have the entire fetch state anyways we can remove the bool!

[the remark this answers is not in the document]
Comment (YannickPferr) [reply internal/pkg/flink/components/table_view.go]: Done!

File: internal/pkg/flink/components/table_view.go (hunk 1 above)
Comment (YannickPferr) [reply internal/pkg/flink/components/table_view.go]: Done!

File: internal/pkg/flink/components/table_view.go (hunk 2 above)
Comment (YannickPferr) [reply internal/pkg/flink/components/table_view.go]: Done!

File: internal/pkg/flink/components/table_info_bar.go [hunk 4]
```diff
@@ -38,8 +38,8 @@ func (t *TableInfoBar) SetLastRefreshTimestamp(lastRefreshTimestamp *time.Time)
 	t.updateInfoBar()
 }
 
-func (t *TableInfoBar) SetFetchState(fetchState types.FetchState) {
-	t.fetchState = fetchState
+func (t *TableInfoBar) SetFetchState(refreshState types.RefreshState) {
```
Comment (Guttz) [inline internal/pkg/flink/components/table_info_bar.go:41]:
```suggestion
func (t *TableInfoBar) SetRefreshState(refreshState types.RefreshState) {
```

File: internal/pkg/flink/components/table_view.go [hunk 5]
```diff
@@ -232,19 +232,36 @@ func (t *TableView) JumpDown() {
...
+	}
+
+	toggleRefreshText := "Play"
+	if refreshState == types.Running {
+		toggleRefreshText = "Pause"
 	}
+	return t.getTableShortcutsForNonCompletedFetchState(toggleTableModeText, toggleRefreshText)
+}
+
+func (t *TableView) getTableShortcutsForCompletedFetchState(toggleTableModeText string) []types.Shortcut {
```
Comment (Guttz) [inline internal/pkg/flink/components/table_view.go:252]:
```suggestion
func (t *TableView) getTableShortcutsForCompletedRefreshState(toggleTableModeText string) []types.Shortcut {
```

**Review by Guttz — APPROVED** [review]

**Pull request merged**